qa/rpc-tests/segwit.py

```diff
@@ -248,20 +248,9 @@ def run_test(self):
         assert(tmpl['transactions'][0]['txid'] == txid)
         assert(tmpl['transactions'][0]['sigops'] == 8)
 
-        print("Verify non-segwit miners get a valid GBT response after the fork")
+        print("Non-segwit miners are not able to use GBT response after activation.")
         send_to_witness(1, self.nodes[0], find_unspent(self.nodes[0], 50), self.pubkey[0], False, Decimal("49.998"))
-        try:
-            tmpl = self.nodes[0].getblocktemplate({})
-            assert(len(tmpl['transactions']) == 1)  # Doesn't include witness tx
-            assert(tmpl['sizelimit'] == 1000000)
-            assert('weightlimit' not in tmpl)
-            assert(tmpl['sigoplimit'] == 20000)
-            assert(tmpl['transactions'][0]['hash'] == txid)
-            assert(tmpl['transactions'][0]['sigops'] == 2)
-            assert(('!segwit' in tmpl['rules']) or ('segwit' not in tmpl['rules']))
-        except JSONRPCException:
-            # This is an acceptable outcome
-            pass
+        assert_raises_jsonrpc(-8, "Support for 'segwit' rule requires explicit client support", self.nodes[0].getblocktemplate, {})
```

Comment:
Thanks @MarcoFalke. I'm not a fan of leaving unexecuted lines in the test code just in case the product code is changed in the future. Re-adding a few asserts in this test case is negligible effort compared to actually adding the functionality to bitcoin core (which would require significant changes to the mining code). 

And if writing 7 assert lines is really too much effort, git history is a far better place to keep this dead code :)